### PR TITLE
Retrieval improvements

### PR DESCRIPTION
**Pull request by whyrusleeping**
Working on improving retrieval.

- Expose utilities to inspect retrieval data on the miner side
- add a `lotus client info` command
- fix the output of `lotus client get-deal`
- make the default output of `lotus client list-deals` more compact
- update to version of go-fil-markets with a number of fixes
- misc other improvements

### CODE REVIEW

**Comment by whyrusleeping** [comment]
@magik6k in order to finish this, I need to resolve this TODO in sector storage: https://github.com/filecoin-project/sector-storage/blob/master/manager.go#L218

File: markets/retrievaladapter/provider.go [hunk 1]
```diff
@@ -39,7 +41,8 @@ func (rpn *retrievalProviderNode) GetMinerWorkerAddress(ctx context.Context, min
 	return mi.Worker, err
 }
 
-func (rpn *retrievalProviderNode) UnsealSector(ctx context.Context, sectorID uint64, offset uint64, length uint64) (io.ReadCloser, error) {
+func (rpn *retrievalProviderNode) UnsealSector(ctx context.Context, sectorID abi.SectorNumber, offset abi.UnpaddedPieceSize, length abi.UnpaddedPieceSize) (io.ReadCloser, error) {
+	fmt.Println("Unsealing Sector: ", sectorID, offset, length)
```
Comment (arajasek) [inline markets/retrievaladapter/provider.go:45]: eh?

File: markets/retrievaladapter/provider.go [hunk 2]
```diff
@@ -57,7 +60,15 @@ func (rpn *retrievalProviderNode) UnsealSector(ctx context.Context, sectorID uin
 
 	r, w := io.Pipe()
 	go func() {
-		err := rpn.sealer.ReadPiece(ctx, w, sid, storiface.UnpaddedByteIndex(offset), abi.UnpaddedPieceSize(length), si.TicketValue, *si.CommD)
+		var commd cid.Cid
+		if si.CommD != nil {
+			commd = *si.CommD
+		}
+		fmt.Println("read piece...")
```
Comment (arajasek) [inline markets/retrievaladapter/provider.go:67]: eh?

File: markets/retrievaladapter/provider.go [hunk 3]
```diff
@@ -57,7 +60,15 @@ func (rpn *retrievalProviderNode) UnsealSector(ctx context.Context, sectorID uin
...
+		err := rpn.sealer.ReadPiece(ctx, w, sid, storiface.UnpaddedByteIndex(offset), length, si.TicketValue, commd) // do we really need to pass a commD here?
+		if err != nil {
+			fmt.Println("read piece errored: ", err)
```
Comment (arajasek) [inline markets/retrievaladapter/provider.go:70]: eh?

File: markets/storageadapter/provider.go [hunk 4]
```diff
@@ -229,6 +230,7 @@ func (n *ProviderNodeAdapter) LocatePieceForDealWithinSector(ctx context.Context
 	if bestSi.State == sealing.UndefinedSectorState {
 		return 0, 0, 0, xerrors.New("no sealed sector found")
 	}
+	// TODO(review) is this offset actually padded?
```
Comment (magik6k) [inline markets/storageadapter/provider.go:233]: Yes, did go through a bunch of levels, and it seems to be used correctly everywhere

```suggestion
```

**Comment by hannahhoward** [comment]
@whyrusleeping so I get the usability reasons for making list-deals more compact. However at this stage in the network, having the default output hide the message makes receiving this output from miners significantly less useful for debugging. I wonder if we should delay this till mainnet so the default output gives us more information.

**Pull request merged**